src/build.rs

```diff
@@ -199,13 +197,8 @@ impl BuildQueue {
         if !pending.is_empty() {
             // Kick off one build, then skip the rest.
             let mut pending = pending.iter();
-            loop {
-                let next = pending.next();
-                let next = match next {
-                    Some(n) => n,
-                    None => break,
-                };
-                if let Ok(_) = next.send(Signal::Build) {
+            while let Some(next) = pending.next() {
```

Comment:
Unfortunately, a `for` loop would consume `pending`, and we need it in another `for`loop afterwards.